### PR TITLE
Changed table layouts and added timings

### PR DESCRIPTION
**Pull request by copaah**
I've changed the table layouts as we discussed monday. Please check if my approach to timing makes sense.

### CODE REVIEW

File: benchmark/results.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+\begin{tabular}{lrr}
+\hline
+ Model     &   MNIST &   Iris \\
+\hline
+ SparseMax &   0.050 &  0.040 \\
```
Comment (AndreasMadsen) [inline benchmark/results.txt]: Can you transpose this, so it is like `timings.txt` I also think the confidence intervals are missing.

File: benchmark/run.py [hunk 2]
```diff
@@ -1,29 +1,133 @@
...
+from sklearn.preprocessing import LabelBinarizer
+
+from tabulate import tabulate
+
+def computeStats(results):
+    n = len(results)
+    mean = round(np.mean(results),2)
+    se = np.std(results)/np.sqrt(n)
+    ci = round(1.96*se,2)
+    return str(mean) + "+-" + str(ci)
```
Comment (AndreasMadsen) [inline benchmark/run.py]: `"$%.2f \\pm %.2f$ % (mean, ci)` will look better

File: benchmark/run.py [hunk 3]
```diff
@@ -1,29 +1,133 @@
+import IPython
```
Comment (AndreasMadsen) [inline benchmark/run.py]: I'm curious, what is this for?

File: benchmark/run.py [hunk 4]
```diff
@@ -1,29 +1,133 @@
...
+    digits = datasets.load_digits()
+    iris = datasets.load_iris()
+
+    # Define list of datasets
+    lDataset = [digits, iris]
+
+    # Define names of datasets to be displayed in .tex table
+    dataset_tex = ["MNIST", "Iris"]
+
+    computeResults(regressors, regressors_tex, lDataset, dataset_tex,
```
Comment (AndreasMadsen) [inline benchmark/run.py]: It's not important, but it would be nice if you would split this file up in `results.py` and `timings.py`. Then it will be easier to read.

File: benchmark/model_test.py [hunk 5]
```diff
@@ -55,15 +55,20 @@ def test_fold(self, fold, train_index, test_index):
...
+            losses.append(result.loss)
+            missrates.append(result.missrate)
+
+        avg_loss, avg_missrate = np.mean(losses),\
+                                 np.mean(missrates)
 
-            # assert prediction scores
-            assert_true(results.loss < loss)
-            assert_true(results.missrate < missrate)
+        return avg_missrate
```
Comment (AndreasMadsen) [inline benchmark/model_test.py]: Yeah, so I would just return `(losses, missrates)` then you can process it later.

File: benchmark/timings.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+\begin{tabular}{lll}
+\hline
+ Dataset   & SparseMax   & SoftMax    \\
```
Comment (AndreasMadsen) [inline benchmark/timings.txt]: The timings table should have all the implementations.

**Review by AndreasMadsen — CHANGES_REQUESTED** [review]
Nice. I will add the GPU stuff thursday.

Also, it looks like `tensorflow_sparsemax/build.mk` got deleted.

**Review by AndreasMadsen — APPROVED** [review]

**Comment by AndreasMadsen** [comment]
Added a lot of dataset from the original paper and setup individual GPU and CPU sparsemax regression models.

Besides that:
- The numpy version was relatively quite fast, because we where transferring all the data in each epoch. There is now a fast mode that fixes that.
- MNIST was not performing very well, I changed the models to use default ADAM
- The new dataset are multi-label, thus we are now reporting Jesen-Shannon divergence instead of missrate. This is similar to the original paper.
- A lot of refactoring to make adding datasets and models easier.

I'm currently running the results on HPC, I will upload the results tomorrow morning if they are fine (no bugs). I have tested I quite thoroughly so I do expect it all to work.

**Comment by AndreasMadsen** [comment]
There is something odd with the softmax+mnist results, but all other results looks fine. I will merge this now and look at the softmax+mnist later.
